### ISSUE TITLE
Unifall flag?

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#17565]
Would it be possible to get a flag that makes `apply`, `rewrite`, etc use the "new" (`refine`) unification rather than the legacy one?  As I understand it, actually swapping to the new unification everywhere is a slow progress with regard to compatibility, but in cases where the new engine behaves better, it would be nice to be able to take advantage of it.

**Comment by mattam82** [comment]
Yes, that's on my TODO, it would be a lot less friction than trying to port everything (and have to replicate in `evarconv.ml` the "misbehaviors" of `unification.ml`)
